src/lib/y2storage/disk_size.rb

```diff
@@ -609,10 +609,48 @@ def floor(unit_size)
     # @example
     #   x = DiskSize.KB(1)   #=> <DiskSize 0.98 KiB (1000)>
     #   x.to_human_string    #=> "0.98 KiB"
-    def to_human_string
+    def to_human_string(round_method = :round)
       return "unlimited" if unlimited?
-      size, unit = human_string_components
-      format("%.2f %s", size, unit)
+      float, unit_s = human_string_components
+      rounded = (float * 100).public_send(round_method) / 100.0
+      # A plain "#{rounded} #{unit_s}" would not keep trailing zeros
+      format("%.2f %s", rounded, unit_s)
+    end
+
+    # A human readable string that does not exceed the exact size.
+    #
+    # If we have 4.999 GiB of space, and ask the user how much of that
+    # should be used, prefilling the "Size" widget with the maximum
+    # rounded up to "5.00 GiB" it will then fail validation.
+    # We must round down.
+    #
+    # @return [String]
+    #
+    # @example
+    #   x = DiskSize.new(4095)  #=> <DiskSize 4.00 KiB (4095)>
+    #   x.to_human_string       #=> "4.00 KiB"
+    #   x.human_floor           #=> "3.99 KiB"
+    #
+    # @see to_human_string
+    # @see human_ceil
```

Comment:
In my opinion, these should be examples of `x.to_human_string(:floor)`. This places the useful information in the method that is just an alias.